lib/layer.js

```
@@ -83,12 +83,18 @@ function Layer (path, options, fn) {
       }
     }
 
-    return pathRegexp.match((opts.strict ? _path : loosen(_path)), {
-      sensitive: opts.sensitive,
-      end: opts.end,
-      trailing: !opts.strict,
-      decode: decodeParam
-    })
+    try {
+      return pathRegexp.match((opts.strict ? _path : loosen(_path)), {
+        sensitive: opts.sensitive,
+        end: opts.end,
+        trailing: !opts.strict,
+        decode: decodeParam
+      });
+    }
+    catch (err) {
+      console.error(err);
```

I’d prefer to merge the message you’re suggesting with the error message, rather than logging the error to the console and then throwing a new one